From d08136f5560d1ce65a781e63ad8b84f0be79cce5 Mon Sep 17 00:00:00 2001
From: Leon Schumacher <leonsch@protonmail.com>
Date: Thu, 23 Jan 2025 09:41:18 +0100
Subject: [PATCH] use temporary file instead of hello.typ

---
 main.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/main.py b/main.py
index e0e849d..114c0f2 100644
--- a/main.py
+++ b/main.py
@@ -4,9 +4,11 @@ import time
 import typst
 import argparse
 from jsonrpcserver import method, serve, Success, Error
+import tempfile
 
 # Path to the .typ file
-file_path = 'hello.typ'
+file = tempfile.NamedTemporaryFile(suffix=".typ")
+file_path = file.name
 compiler = typst.Compiler(file_path)
 default_preamble = ""
 
-- 
2.47.0

