From 50c75a7d6fa57b27e0e090583fd8e25cba369592 Mon Sep 17 00:00:00 2001
From: Leon Schumacher <leonsch@protonmail.com>
Date: Thu, 23 Jan 2025 09:27:56 +0100
Subject: [PATCH 1/2] SafeConfigParser has been removed

---
 versioneer.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/versioneer.py b/versioneer.py
index 64fea1c..3aa5da3 100644
--- a/versioneer.py
+++ b/versioneer.py
@@ -339,9 +339,9 @@ def get_config_from_root(root):
     # configparser.NoOptionError (if it lacks "VCS="). See the docstring at
     # the top of versioneer.py for instructions on writing your setup.cfg .
     setup_cfg = os.path.join(root, "setup.cfg")
-    parser = configparser.SafeConfigParser()
+    parser = configparser.ConfigParser()
     with open(setup_cfg, "r") as f:
-        parser.readfp(f)
+        parser.read_file(f)
     VCS = parser.get("versioneer", "VCS")  # mandatory
 
     def get(parser, name):
-- 
2.47.0

