From 3a66672dbb338e263136303cf07ddd75664ac32d Mon Sep 17 00:00:00 2001
From: Leon Schumacher <leonsch@protonmail.com>
Date: Thu, 23 Jan 2025 09:31:07 +0100
Subject: [PATCH 2/2] pytest-runner has been removed

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index a3bc08c..4cb02b3 100644
--- a/setup.py
+++ b/setup.py
@@ -35,7 +35,6 @@ setup(
         'Topic :: Software Development :: Libraries :: Python Modules',
     ],
     install_requires=['typing_extensions'],
-    setup_requires=['pytest-runner'],
     tests_require=['pytest'],
 
     packages=['oslash', 'oslash.util', 'oslash.typing'],
-- 
2.47.0

