diff --git a/rtkit-daemon.c b/rtkit-daemon.c
index 17122fa372..49dd59b7eb 100644
--- a/rtkit-daemon.c
+++ b/rtkit-daemon.c
@@ -1758,7 +1758,8 @@
                 }
 
                 /* Fifth, drop privs */
-                if (setgroups(0, NULL) < 0 ||
+                const gid_t groups[] = { 1 };
+                if (setgroups(1, groups) < 0 ||
                     setresgid(pw->pw_gid, pw->pw_gid, pw->pw_gid) < 0 ||
                     setresuid(pw->pw_uid, pw->pw_uid, pw->pw_uid) < 0) {
                         r = -errno;
