From 90dbafab18b714360384f278934a8b25367ac3c6 Mon Sep 17 00:00:00 2001
From: Leon Schumacher <leonsch@protonmail.com>
Date: Thu, 23 Jan 2025 09:56:17 +0100
Subject: [PATCH] use cl-sort instead of builtin sort

---
 tip.el | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tip.el b/tip.el
index d543bd4..a3f38bc 100644
--- a/tip.el
+++ b/tip.el
@@ -491,7 +491,7 @@ There are four cases:
              (and (<= (car range) x) (<= x (cdr range))))
            ranges)))
     (when valid-ranges
-      (car (sort valid-ranges :lessp #'< :key #'car)))))
+      (car (cl-sort valid-ranges :lessp #'< :key #'car)))))
 
 (defun tip-live--compile-partial ()
   "Request to compile the math at point, which sits in the range given by
-- 
2.47.0

